### PR TITLE
[chore] optimize `OperationStructure` construction time

### PR DESCRIPTION
**Pull request by gee-senbong**
## Description

This PR removes pydantic BaseModel from `OperationStruction` to optimize operation structure extraction runtime.

## Performance Results

command: `sudo -E py-spy record -o profile.svg --  pytest -vv tests/integration/api/test_event_view_operations.py -k test_datetime_operations --source-types=snowflake`

main:
profile_1807

current branch:
profile_1806

Time to save 204 new features drops from 12min to 11.5min.
Time to save a feature list from 204 created features & the feature list drops from 40sec to 35sec.

## Related Issue

https://github.com/featurebyte/featurebyte/pull/1415

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/query_graph/node/base.py [hunk 1]
```diff
@@ -262,7 +262,16 @@ def derive_node_operation_info(
...
             or operation_info.is_time_based,
         }
-        return OperationStructure(**{**operation_info.dict(), **update_args})
+        constructor_args = {
+            "columns": operation_info.columns,
+            "aggregations": operation_info.aggregations,
+            "output_type": operation_info.output_type,
+            "output_category": operation_info.output_category,
+            "row_index_lineage": operation_info.row_index_lineage,
+            "is_time_based": operation_info.is_time_based,
```
Comment (YS-L) [inline featurebyte/query_graph/node/base.py:271]: Can `is_time_based` be set as

```
any(input_.is_time_based for input_ in inputs) or operation_info.is_time_based
```

directly? That seems clearer and then `update_args` is unnecessary.

Comment (gee-senbong) [reply featurebyte/query_graph/node/base.py:271]: Good point, will update this part to make it cleaner. Maybe i should pass the parameter directly to the `OperationStructure` without constructing a dictionary here.

**Review by YS-L — APPROVED** [review]
LGTM

**Review by kchua78 — APPROVED** [review]
Not sure how to understand the charts - but it's great to improve the serde ops efficiency here 👍

**Pull request merged**